lightwood/api/data_source.py

```diff
@@ -229,72 +239,42 @@ def prepare_encoders(self):
             from the training dataset.
         """
         input_encoder_training_data = {'targets': []}
-        self.out_types = []
-
-        for feature_set in ['output_features', 'input_features']:
-            for feature in self.configuration[feature_set]:
-                column_name = feature['name']
-                config = self.get_column_config(column_name)
+        self.out_types = [config['type'] for config in self.configuration['output_features']]
 
-                args = [self.get_column_original_data(column_name)]
+        for config in self.configuration['output_features']:
```

Comment:
Since you've removed most of the boilerplate and separate this into two loops you might as well make this a separate function that returns `input_encoder_training_data` + stets `self.out_type`, I think it would make the whole thing make even more sense. 

But just a stylistic thing, go with whatever you think is best.